test/index.js

```diff
@@ -0,0 +1,102 @@
+const test = require('ava');
+const { ServiceBroker } = require('moleculer');
+const fetch = require('node-fetch');
+const MongoHealthcheck = require('..');
+const HealthMiddleware = require('@r2d2bzh/moleculer-healthcheck-middleware');
+const DbService = require('moleculer-db');
+const MongoDBAdapter = require('moleculer-db-adapter-mongo');
+
+const event = (emitter, eventName) =>
+  new Promise((resolve) => emitter.once(eventName, resolve));
+
+const startBroker = async (t, services, healthCheckOpts = {}) => {
+  const broker = new ServiceBroker({
+    middlewares: [HealthMiddleware({
+      liveness: {
+        createChecker: MongoHealthcheck.createLivenessChecker
+      },
+      port: 0,
+      ...healthCheckOpts
+    })],
+  });
+  services.forEach(service => {
+    broker.createService(service);
+  });
+  await broker.start();
+
+  let port;
+  if (!broker.healthcheck.port) {
+    port = await event(broker.healthcheck, 'port');
+  } else {
+    port = broker.healthcheck.port;
+  }
+
+  t.context.broker = broker;
+  t.context.healthport = port;
+}
+
+const mockAdapter = (t) => {
+  return {
+    init: () => new Promise(resolve => {resolve(true)}),
+    connect: () => new Promise(resolve => {resolve(true)}),
+    client: {
+      db: () => {
+        return {
+          topology: {
+            isConnected: () => { t.fail('This should not be called') }
+          }
+        }
+      }
+    }
+  }
+};
+
+test.afterEach.always(async (t) => {
+  if(t.context.broker) {
+    await t.context.broker.stop();
+  }
+});
+
+test('healthcheck answers a 200 when mongo is up', async (t) => {
+  await startBroker(t, [
+    {
+      name: 'adapterMongo',
+      mixins: [DbService],
+      adapter: new MongoDBAdapter('mongodb://localhost'),
+      collection: 'posts',
+    }
+  ]);
+  const port = t.context.healthport;
+  const response = await fetch(`http://127.0.0.1:${port}/live`).catch(error => error);
+  t.deepEqual(response.status, 200);
+});
+
+test('healthcheck answers a 503 when no adapter is found', async (t) => {
```

Comment:
I am not sure I understand the goal of this test the right way.
Does this mean that when there is no db adapter to check the services registered to this broker will be considered as dead?

Reply:
The goal is to ensure that the liveness check is in error when no adapters are found.
The issue in this case is that in the checker we just give an error message to the next() function.

In this case, the liveness is not working so it passes an error when calling next(). Every error sent by the checker will be sent as 503 by the [moleculer-healthcheck-middleware](https://github.com/r2d2bzh/moleculer-healthcheck-middleware).

If needed, some work needs to be done on [moleculer-healthcheck-middleware](https://github.com/r2d2bzh/moleculer-healthcheck-middleware) on error handling. Passing the message and the error code to next() might be better for this case.